### DISCUSSION TITLE
cleanup with out reboot

### DISCUSSION THREAD

**Discussion by Lord1340** [cregx/mdt-cleanup-harddrive-winpe#4] [Ideas]
Hi,
Why not compile a cleanup.exe with out reboot?
In our case we don´t get so far in the process so a reboot is necessary.
Just a thougt.

Regards and a good tool

**Comment by Lord1340** [comment]
Hi,
Why not compile a cleanup.exe with out reboot?
In our case we don´t get so far in the process so a reboot is necessary.
Or you can have a counter countig down from xx sec, and if not Cleanup key is pressed the skip key should be pressed after time out.

Regards and a good tool

**Reply by cregx** [reply]
Hello,

You have not contacted me again. Have you programmed and compiled the solution with the restart - as I suggested it to you - yourself?

I would be very reluctant to have such a solution fixed, because it would give the impression that you can do without the restart. As already written, my experiences have shown that a restart is always necessary.

Many greetings

**Comment by cregx** [comment]
Hello,

Thank you very much for your suggestions. The first version of Cleanup did indeed not automatically restart. However, this proved to be counterproductive, as in some cases, the subsequent installation process failed. I observed at the time that this problem did not occur if a system reboot was performed after the disk cleanup.

It should be possible to compile Cleanup without a reboot. Either do this yourself or wait until I have programmed an appropriate solution, e.g., your suggestion with the countdown or a call parameter that checks whether a reboot should be performed when Cleanup is called. The second solution (call parameter) seems more practical to me because it also takes into account the concerns of my environment.

I am glad that you like Cleanup.

Regards

**Comment by Lord1340** [comment]
I haven’t did that yet, but i have a co worker i belive can compile a version //Tommy Skickat från min IPhone 8 29 apr. 2023 kl. 19:38 skrev cregx ***@***.***>:﻿
Hello,
You have not contacted me again. Have you programmed and compiled the solution with the restart - as I suggested it to you - yourself?
I would be very reluctant to have such a solution fixed, because it would give the impression that you can do without the restart. As already written, my experiences have shown that a restart is always necessary.
Many greetings

**Reply by cregx** [reply]
I have considered your suggestion and will work out a solution in the form of an option (CheckBox). This will have it checked by default and tell the user that Cleanup will automatically restart the system when cleanup is complete.
If the user unchecks the box, no restart will occur. I hope this is to your liking.

**Comment by Lord1340** [comment]
Yes!I think this is a perfect solution!//Tommy Skickat från min IPhone 8 1 maj 2023 kl. 08:32 skrev cregx ***@***.***>:﻿
I have considered your suggestion and will work out a solution in the form of an option (CheckBox). This will have it checked by default and tell the user that Cleanup will automatically restart the system when cleanup is complete.
If the user unchecks the box, no restart will occur. I hope this is to your liking.

**Comment by cregx** [comment]
Thank you to @Lord1340  for your idea suggestion, which has now been implemented. I really appreciate your contribution and am glad that I was able to implement your idea. Your contribution was valuable in improving the software. I hope you find the new functionality useful. Thank you again for your valuable contribution.

If you have any further questions or suggestions, please don't hesitate to let me know.

Best regards

**Reply by cregx** [reply]
https://github.com/cregx/mdt-cleanup-harddrive-winpe/releases/tag/v.1.1.6
